app-admin/logstash-bin/files/logstash-plugin.eselect

```diff
@@ -0,0 +1,77 @@
+# -*-eselect-*-  vim: ft=eselect
+# Copyright 2005-2017 Gentoo Foundation
+# Distributed under the terms of the GNU GPL version 2 or later
+
+DESCRIPTION="Manage user installed plugins of logstash"
+MAINTAINER="hydrapolic@gmail.com"
+
+PLUGINS="/var/lib/logstash/plugins-list.txt"
+LS_INSTALL_DIR="/opt/logstash"
+
+ls_plugin_wrapper() {
+  JARS_SKIP=true "${LS_INSTALL_DIR}/bin/logstash-plugin" "$1" "$2"
+}
+
+### add action
+describe_add() {
+  echo "Add plugin"
+}
+
+describe_add_parameters() {
+  echo "<plugin>"
+}
+
+do_add() {
+  local x
+  for x; do
+    echo "$x" >> ${PLUGINS} || die
+    ls_plugin_wrapper install "$x"
```

Comment:
I've swapped the install/record logic and since the wrapper dies on failures, if you try to install foobar, it will not continue. If you have happen to have some obsolete entry while doing delete, we first delete from the text file, then remove from logstash.